### PR TITLE
Add tests for translated key order and some cases of duplicate keys.

### PR DESCRIPTION
**Pull request by sfadschm**
This adds a test method to check whether the order of the translated keys is the same as in the main repo.
At the same time, this also protects against duplicate key definitions in translated file, but not all possible cases.

Example that will trigger an error:

    'key1' => 'val1',
    'key2' => 'val2',
    'key1' => 'val3',

Example that will not trigger an error:

    'key1' => 'val1',
    'key1' => 'val3',
    'key2' => 'val2',

Example output for current `it` locale:

    Failed asserting that the translated language keys in "it" locale have the same order as the original keys in the main repository. This can also be caused by duplicate key definitions. First mismatch per file:
    Migrations:
      - 26 => 'migSeeder'
      + 26 => 'on'  
    Validation:
     - 26 => 'matches'
     + 26 => 'string'
    Failed asserting that an array is empty.

I would think the other cases of duplicate keys are pretty obvious to see when editing a translation file and should not occur too often. If we wanted to check them too, we would need to `file_get_contents` the language files and compare the line on which each key is defined (or the total number of lines in the file). This would of course also require us to have consistent comments and blank lines in the files (which is not a bad thing in my opinion).

### CODE REVIEW

File: tests/Language/AbstractTranslationTestCase.php [hunk 1]
```diff
@@ -378,4 +424,27 @@ private function translationSets(?string $locale = null): array
...
+	 * @return sting
+	 */
+	private function array_diff_order(array $array1, array $array2): string
+	{
+		foreach($array1 as $key1 => $val1)
+		{
+			$val2 = $array2[$key1] ?? null;
+			if($val2 && $val2 !== $val1)
+			{
+				return "  - $key1 => '$val1' \n  + $key1 => '$val2'";
```
Comment (paulbalandan) [inline tests/Language/AbstractTranslationTestCase.php:444]:
```suggestion
				return "  - $key1 => '$val1'\n  + $key1 => '$val2'";
```

File: tests/Language/AbstractTranslationTestCase.php [hunk 2]
```diff
@@ -378,4 +424,27 @@ private function translationSets(?string $locale = null): array
...
 	}
+
+	/**
+	 * Checks if the order of array keys in two arrays is equal.
+	 *
+	 * @param array $array1
+	 * @param array $array2
+	 *
+	 * 
+	 * @return sting
```
Comment (paulbalandan) [inline tests/Language/AbstractTranslationTestCase.php:435]:
```suggestion
	 * @return string
```

File: tests/Language/AbstractTranslationTestCase.php [hunk 3]
```diff
@@ -378,4 +424,27 @@ private function translationSets(?string $locale = null): array
 
 		return $sets;
 	}
+
+	/**
+	 * Checks if the order of array keys in two arrays is equal.
+	 *
+	 * @param array $array1
+	 * @param array $array2
```
Comment (paulbalandan) [inline tests/Language/AbstractTranslationTestCase.php:432]:
```suggestion
	 * @param array<string, string> $array1
	 * @param array<string, string> $array2
```

**Review by paulbalandan — CHANGES_REQUESTED** [review]

File: tests/Language/AbstractTranslationTestCase.php [hunk 4]
```diff
@@ -274,6 +274,52 @@ abstract class AbstractTranslationTestCase extends TestCase
...
+					$diffs[substr($file, 0, -4)] = $diff;
+				}
+			}
+		}
+
+		// Prepare diffs for output
+		array_walk($diffs, fn(&$item, $key) => $item = $key.":\n".$item);
+
+		self::assertEmpty($diffs, sprintf(
+			'Failed asserting that the translated language keys in "%s" locale have the same order as the original keys in the main repository. This can also be caused by duplicate key definitions. First mismatch per file:'."\n".'%s',
```
[the remark this answers is not in the document]
Comment (sfadschm) [reply tests/Language/AbstractTranslationTestCase.php:316]: Will make more precise. Do you mean merging into a single line message?

File: tests/Language/AbstractTranslationTestCase.php [hunk 5]
```diff
@@ -274,6 +274,52 @@ abstract class AbstractTranslationTestCase extends TestCase
...
+				// Store result if failed
+				if($diff !== '')
+				{
+					$diffs[substr($file, 0, -4)] = $diff;
+				}
+			}
+		}
+
+		// Prepare diffs for output
+		array_walk($diffs, fn(&$item, $key) => $item = $key.":\n".$item);
```
[the remark this answers is not in the document]
Comment (sfadschm) [reply tests/Language/AbstractTranslationTestCase.php]: Removed the need for `array_walk`.

[the remark this answers is not in the document]
Comment (sfadschm) [reply tests/Language/AbstractTranslationTestCase.php]: Refactored the message to 

    Failed asserting that the translated language keys in "it" locale are ordered correctly. "Migrations.php" index "26" is "on", should be "migSeeder". "Validation.php" index "26" is "string", should be "matches".
Failed asserting that an array is empty.

Is this more like expected?

File: tests/Language/AbstractTranslationTestCase.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (paulbalandan) [reply tests/Language/AbstractTranslationTestCase.php]: I mean here is you used concat on all strings which is useless.

File: tests/Language/AbstractTranslationTestCase.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (paulbalandan) [reply tests/Language/AbstractTranslationTestCase.php]: I would like the second sentence to be the earlier `git diff` format. That's visually better, I think

File: tests/Language/AbstractTranslationTestCase.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfadschm) [reply tests/Language/AbstractTranslationTestCase.php]: Oh right, I thought we were going with `''` for strings, but `""` is required for newlines to be interpreted. Will change.

**Review by paulbalandan — APPROVED** [review]
I have some visual adjustments but I'll just do that myself in a later PR.

**Comment by MGatner** [comment]
Very nice work, thank you team.
